configure.ac

```
@@ -1175,7 +1175,7 @@ dnl Do not change "-I/usr/include" to "-isystem /usr/include" because that
 dnl is not necessary (/usr/include is already a system directory) and because
 dnl it would break GCC's #include_next.
 AC_DEFUN([SUPPRESS_WARNINGS],
-         [$(echo $1 |${SED} -E -e 's/(^| )-I/\1-isystem /g' -e 's;-isystem /usr/include([/ ]|$);-I/usr/include\1;g')])
+         [[$(echo $1 |${SED} -E -e 's/(^| )-I/\1-isystem /g' -e 's;-isystem /usr/include([/ ]|$);-I/usr/include\1;g')]])
```

@luke-jr 

Does
```suggestion
         [[$(echo $1 |${SED} -E -e 's/(^| )-I/\1-isystem /g' -e 's;-isystem /usr/include( |$);-I/usr/include\1;g')]])
```
work for you?
```

---

Yes, but we should probably still quote the command properly.

I don't understand why `-isystem` breaks things for some paths... Hopefully @vasild can comment

---

If `-isystem /usr/include` is used with gcc, then the compilation would fail because that somehow breaks the `#include_next <...>` directives used by gcc headers themselves. For example, on my system with gcc 10:

`t.cc`:
```cpp
#include <iostream>

int main(int, char**)
{
    return 0;
}
```

```
g++10 -isystem /usr/include t.cc -c -o t.o
In file included from /usr/local/lib/gcc10/include/c++/ext/string_conversions.h:41,
                 from /usr/local/lib/gcc10/include/c++/bits/basic_string.h:6557,
                 from /usr/local/lib/gcc10/include/c++/string:55,
                 from /usr/local/lib/gcc10/include/c++/bits/locale_classes.h:40,
                 from /usr/local/lib/gcc10/include/c++/bits/ios_base.h:41,
                 from /usr/local/lib/gcc10/include/c++/ios:42,
                 from /usr/local/lib/gcc10/include/c++/ostream:38,
                 from /usr/local/lib/gcc10/include/c++/iostream:39,
                 from t.cc:1:
/usr/local/lib/gcc10/include/c++/cstdlib:75:15: fatal error: stdlib.h: No such file or directory
   75 | #include_next <stdlib.h>
      |               ^~~~~~~~~~
compilation terminated.
*** Error code 1
```